### ISSUE TITLE
Vagrant chef cleanup fails with the latest Chef Infra client version 17.0.242 with "bash: line 4: knife: command not found"

### ISSUE THREAD

**Issue by canoenc** [hashicorp/vagrant#12338]
With the latest Chef Infra Client they (Opscode) have removed the **knife** command as well as the **chef** command.
Those commands are only available in Chef Workstation.
Check their Release Notes: https://github.com/chef/chef/blob/master/RELEASE_NOTES.md

So if you use Chef provisioning and then Chef cleanup in your Vagrantfile you get failures due to the knife command not found as Vagrant runs the knife node delete and knife client delete commands using bash on the vagrant guest virtual machine that was provisioned.

Vagrantfile chef cleanup
-------------------------

              **chef.delete_node = true
              chef.delete_client = true**

--------------------------------------------------------------------------------------------------------------------------

------------------------------
Vagrant guest virtual machine
------------------------------

$ vagrant ssh
Last login: Thu Apr 29 19:03:44 2021 from 10.0.2.2
$ sudo su -
Last login: Thu Apr 29 15:03:48 EDT 2021 on pts/0
chef-client --version
Chef Infra Client: 17.0.242**
 chef --version
**-bash: chef: command not found**
knife
**-bash: knife: command not found**
exit
logout
exit
logout
Connection to 127.0.0.1 closed.

--------------------------------------------------------------------------------------------------------------------------

--------------------------
**Vagrant host** machine
--------------------------

vagrant destroy -f
==> default: Running cleanup tasks for 'chef_client' provisioner...
==> default: Deleting node "apache.acme.com" from Chef server...
==> default: There were errors removing the node from the Chef Server:
==> default:
**==> default: bash: line 4: knife: command not found**
==> default:
==> default:
==> default: Vagrant will continue destroying the virtual machine, but you may need
==> default: to manually delete the node from the Chef Server!
==> default: Deleting client "apache.acme.com" from Chef server...
==> default: There were errors removing the client from the Chef Server:
==> default:
**==> default: bash: line 4: knife: command not found**
==> default:
==> default:
==> default: Vagrant will continue destroying the virtual machine, but you may need
==> default: to manually delete the client from the Chef Server!
==> default: Forcing shutdown of VM...
==> default: Destroying VM and associated drives...

**Environment (please complete the following information):**

Really does matter but anyway.

- Host OS: CentOS 7.9
- Kernel version (for Linux host): [run `uname -a`]
- Vagrant version:  2.2.15
- Vagrant provider:
  - For VirtualBox:
    - VirtualBox version: 6.1.20r143896
  
**Additional information**

Add any other information about the issue here (console log, ...).

**Comment by lamont-granquist** [comment]
> With the latest Chef Infra Client they (Opscode) have removed the knife command as well as the chef command.

The `chef` command has never been shipped with the client and always been part of chef-workstation or chefdk.

I think this bug report is more narrowly about the removal of the `knife` command from Chef Client 17 and its extraction into its own gem.

**Comment by indygwyn** [comment]
See my comment on #12346

TLDR, set your chef.product to be `chef-workstation` (which includes knife and chef commands, I'm not sure this shouldn't be the default with all the chef provisioners now that knife is no longer included with Infra Client

**Comment by fuegas** [comment]
@indygwyn, just like "the fix" for the knife PTY default, I've created a script for my team that fixes the detection to something usable for us: https://gist.github.com/fuegas/90b933784a063c63f35cd13f6a4521ed

We do not need chef-workstation to be installed as we're testing chef (zero) runs, so changing te detection to chef-client instead of knife is good enough to see that the correct package is installed.

The second change made is for vagrant-omnibus, which does not detect the version of Chef correctly. For instance, if you want to use `17.4.38` the installation sees the following output from `chef-solo -v`:
```
Chef Infra Client: 17.4.38
```
Which can't be parsed by `version_match = data.match(/^Chef: (.+)/)`, so we changed the match to `version_match = data.match(/^Chef[^:]*: (.+)/)`. This prevent a reinstall of Chef _every_ provision run of Vagrant.

**Comment by 111andre111** [comment]
@canoe-nc @fuegas 
Would you mind to test with most recent 2.2.19 and no other fixes?
Maybe https://github.com/hashicorp/vagrant/pull/12555 helps here.

**Comment by fuegas** [comment]
@111andre111 : I did not see it on brew available yet, but when it is I will test it without manual fixes.

**Comment by 111andre111** [comment]
Cannot reproduce not being available:
`brew update vagrant` immediately updated to version 2.2.19.
^^ @fuegas

**Comment by fuegas** [comment]
It did not show in my upgrades this morning, but I was already using it 😅 After reverting the "fixes" it works like a charm 👍

**Comment by 111andre111** [comment]
@canoe-nc Cool, then I assume this one can be closed now as well  hopefully for you too 👍
